cli/command/stack/remove.go

```diff
@@ -39,7 +39,17 @@ func runRemove(dockerCli command.Cli, opts removeOptions) error {
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	var errs []string
+	var (
+		errs    []string
+		secrets []swarm.Secret
+		configs []swarm.Config
+	)
+
+	version, err := client.ServerVersion(ctx)
```

Comment:
Ok, so I tried using this, but ran into issues. the `ClientVersion()` is [returning `cli.version`](https://github.com/docker/cli/blob/4adf701567e4e64226e2e3622375b78a612b2198/vendor/github.com/docker/docker/client/client.go#L236-L242)

`cli.version` is only set by [`NegotiateAPIVersionPing()`](https://github.com/docker/cli/blob/4adf701567e4e64226e2e3622375b78a612b2198/vendor/github.com/docker/docker/client/client.go#L263-L266), and [`NewClient()`](https://github.com/docker/cli/blob/4adf701567e4e64226e2e3622375b78a612b2198/vendor/github.com/docker/docker/client/client.go#L191-L200), both of which are not used for our fakeClient.

I decided to use the "old" approach for now, but we should look into this, as it's only an issue for testing

Perhaps you have a trick up your sleeve to make it work though :smile: